contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -0,0 +1,151 @@
+    UpfrontSubscriptionWithEncryptorsCap public immutable subscription;
+
+    /**
+     * @notice Emitted when an administrator cap is set
+     * @param ritualId The ID of the ritual
+     * @param _address The address of the administrator
+     * @param cap The cap value
+     */
+    event AdministratorCapSet(uint32 indexed ritualId, address indexed _address, uint256 cap);
+
+    /**
+     * @notice Sets the coordinator and subscription contracts
+     * @dev The coordinator and subscription contracts cannot be zero addresses
+     * @param _coordinator The address of the coordinator contract
+     * @param _subscription The address of the subscription contract
+     */
+    constructor(
+        Coordinator _coordinator,
+        UpfrontSubscriptionWithEncryptorsCap _subscription
+    ) GlobalAllowList(_coordinator) {
+        require(address(_coordinator) != address(0), "Coordinator cannot be the zero address");
+        require(address(_subscription) != address(0), "Subscription cannot be the zero address");
+        subscription = _subscription;
+    }
+
+    /**
+     * @notice Checks if the sender is the authority of the ritual
+     * @param ritualId The ID of the ritual
+     */
+    modifier onlyCohortAuthority(uint32 ritualId) {
+        require(
+            coordinator.getAuthority(ritualId) == msg.sender,
+            "Only cohort authority is permitted"
+        );
+        _;
+    }
+
+    /**
+     * @notice Checks if the sender has allowance to set authorizations
+     * @dev This function overrides the canSetAuthorizations modifier in the GlobalAllowList contract
+     * @param ritualId The ID of the ritual
+     */
+    modifier canSetAuthorizations(uint32 ritualId) override {
+        require(getAllowance(ritualId, msg.sender) > 0, "Only administrator is permitted");
+        _;
+    }
+
+    /**
+     * @notice Returns the allowance of an administrator for a ritual
+     * @param ritualId The ID of the ritual
+     * @param admin The address of the administrator
+     * @return The allowance of the administrator
+     */
+    function getAllowance(uint32 ritualId, address admin) public view returns (uint256) {
+        return allowance[LookupKey.lookupKey(ritualId, admin)];
+    }
+
+    /**
+     * @dev This function is called before the setAuthorizations function
+     * @param ritualId The ID of the ritual
+     * @param addresses The addresses to be authorized
+     * @param value The authorization status
+     */
+    function _beforeSetAuthorization(
+        uint32 ritualId,
+        address[] calldata addresses,
+        // TODO: Currently unused, remove?
+        // solhint-disable-next-line no-unused-vars
+        bool value
+    ) internal view override {
+        for (uint256 i = 0; i < addresses.length; i++) {
+            require(
+                authActions[ritualId] <
+                    subscription.authorizationActionsCap(ritualId, addresses[i]),
+                "Authorization cap exceeded"
```

Comment:
I may be misunderstanding, but this logic is a little unclear to me:
- Should we only be checking caps if `value` is `True` i.e. we are authorizing? If deauthorizing (i.e. value is False), there are no caps.
- I'm unclear what we are checking in the `require` statement. `canSetAuthorizations` already checks whether the sender is a valid admin for the ritual - great, but I assumed here you need to check 2 things:
   1. the remaining allowance for the admin making the authorization request (i.e. sender) allows for the set of encryptor addresses to be authorized, so:
   ```solidity
   require("getAllowance(ritualId, msg.sender) >= addresses.length", "Administrator cap exceeded")
   ```
   2. That the total cap across all administrators for a specific ritual is not being exceeded, so:
   ```solidity
   require("authActions[ritualId] >= addresses.length", "Ritual-wide cap exceeded")
   ```

Reply:
My calls may be wrong above ^, but the pseudocode I'm looking for is:
i. remaining allowance for administrator to be >= the number of encryptor addresses to authorize
ii. remaining actions for ritual >= the number of encryptor addresses to authorize

Reply:
Just to give more context, I wasn't sure how to use `value` here: Should setting authorization to `false` (so, removing authorization) count towards billable actions? We may treat adding and removing authorization differently, but I think it's application-specific (the frequency of actions, but also the application-specific meaning of authorizing/deauthorizing)